src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -95,6 +95,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
       return;
     }
 
+    if (!conversation.isNormalConversation()) {
+      String user = (String) request.getSession().getAttribute("user");
+
+      // user is not logged in, redirect them to login page
```

Comment:
Maybe move this into a `canAccess(String user)` method

Reply:
Hi Vasu, I just tried to work on this suggestion but ran into some problems. 
![image](https://user-images.githubusercontent.com/11599574/42137577-d8d33496-7d2b-11e8-8ad9-68b1a2d59a32.png) Here's what the code I tried refactoring out looks like, but the problem is that I actually redirect to different pages according to what level of access they have (i.e. if not logged in, they cant access so I return false but it should redirect to /login, if not part of the convo then redirect to /conversations). I thought about changing the return type to a String to represent the page to redirect them to (a non null string means that they cant access the convo, and then redirect them to the returned page in ChatServlet, and a null string means that they can access) but that seems a confusing to treat the null return value as a "yes, the User can access the conversation". Do you have any suggestions as to how I should go about doing this?

Reply:
Yeah, I'd stay away from assigning any semantic meaning to a `null` value in a method response. You can leave it the way it is or pass the `Response` object to your method (which does the redirect) an have it return a boolean value, if the value is false, simply don't proceed

Reply:
Thanks for the help Vasu, I just committed the refactored code and added two ChatServlet tests into the pull request!